python.d/unbound.chart.py

```diff
@@ -0,0 +1,167 @@
+            ['reqlist_exceeded', 'Overruns', 'absolute', 1, 1],
+            ['reqlist_current', 'Current Size', 'absolute', 1, 1],
+            ['reqlist_user', 'User Requests', 'absolute', 1, 1]
+        ]
+    },
+    'recursion': {
+        'options': [None, 'Recursion Timings', 'seconds', 'Unbound', 'unbound.recursion', 'line'],
+        'lines': [
+            ['recursive_avg', 'Average', 'absolute', 1, PRECISION],
+            ['recursive_med', 'Median', 'absolute', 1, PRECISION]
+        ]
+    }
+}
+
+# These get added too if we are told to use extended stats.
+EXTENDED_ORDER = ['cache']
+
+EXTENDED_CHARTS = {
+    'cache': {
+        'options': [None, 'Cache Sizes', 'items', 'Unbound', 'unbound.cache', 'stacked'],
+        'lines': [
+            ['cache_message', 'Message Cache', 'absolute', 1, 1],
+            ['cache_rrset', 'RRSet Cache', 'absolute', 1, 1],
+            ['cache_infra', 'Infra Cache', 'absolute', 1, 1],
+            ['cache_key', 'DNSSEC Key Cache', 'absolute', 1, 1],
+            ['cache_dnscss', 'DNSCrypt Shared Secret Cache', 'absolute', 1, 1],
+            ['cache_dnscn', 'DNSCrypt Nonce Cache', 'absolute', 1, 1]
+        ]
+    }
+}
+
+# This maps the Unbound stat names to our names and precision requiremnets.
+STAT_MAP = {
+    'total.num.queries_ip_ratelimited': ('ratelimit', 1),
+    'total.num.cachehits': ('cachehit', 1),
+    'total.num.cachemiss': ('cachemiss', 1),
+    'total.num.zero_ttl': ('expired', 1),
+    'total.num.prefetch': ('prefetch', 1),
+    'total.num.recursivereplies': ('recursive', 1),
+    'total.requestlist.avg': ('reqlist_avg', 1),
+    'total.requestlist.max': ('reqlist_max', 1),
+    'total.requestlist.overwritten': ('reqlist_overwritten', 1),
+    'total.requestlist.exceeded': ('reqlist_exceeded', 1),
+    'total.requestlist.current.all': ('reqlist_current', 1),
+    'total.requestlist.current.user': ('reqlist_user', 1),
+    'total.recursion.time.avg': ('recursive_avg', PRECISION),
+    'total.recursion.time.median': ('recursive_med', PRECISION),
+    'msg.cache.count': ('cache_message', 1),
+    'rrset.cache.count': ('cache_rrset', 1),
+    'infra.cache.count': ('cache_infra', 1),
+    'key.cache.count': ('cache_key', 1),
+    'dnscrypt_shared_secret.cache.count': ('cache_dnscss', 1),
+    'dnscrypt_nonce.cache.count': ('cache_dnscn', 1)
+}
+
+
+class Service(SocketService):
+    def __init__(self, configuration=None, name=None):
+        # The unbound control protocol is always SSL encapsulated
+        # unless it's used over a UNIX socket, so enable SSL _before_
+        # doing the normal SocketService initialization.
+        configuration['ssl'] = True
+        self.port = 8935
+        SocketService.__init__(self, configuration, name)
+        self.ext = self.configuration.get('extended', None)
+        self.ubconf = self.configuration.get('ubconf', None)
+        self.order = ORDER
+        self.definitions = CHARTS
```

Comment:
Since `order`/`definitions` can be modified you have to copy `ORDER` and `CHARTS`
(`from copy import deepcopy` for `CHARTS`)